www/class/centreonStatistics.class.php

```diff
@@ -137,6 +138,138 @@ public function getPlatformTimezone()
+                ];
+            } elseif ($row['type'] === 'openid') {
+                $authenticationConditions = $customConfiguration['authentication_conditions'];
+                $groupsMapping = $customConfiguration['groups_mapping'];
+                $rolesMapping = $customConfiguration['roles_mapping'];
+                $data['openid'] = [
+                    'is_forced' => $row['is_forced'] ? true : false,
+                    'authenticationConditions' => [
+                        'is_enabled' => $authenticationConditions['is_enabled'] ? true : false,
+                        'trusted_client_addresses'
+                            => isset($authenticationConditions['trusted_client_addresses'])
+                            ? count($authenticationConditions['trusted_client_addresses']) : 0,
+                        'blacklist_client_addresses'
+                            => isset($authenticationConditions['blacklist_client_addresses'])
+                            ? count($authenticationConditions['blacklist_client_addresses']) : 0,
+                        'authorized_values' => isset($authenticationConditions['authorized_values'])
+                            ? count($authenticationConditions['authorized_values']) : 0
+                    ],
+                    'groups_mapping' => [
+                        'is_enabled' => $groupsMapping['is_enabled'] ? true : false,
+                        'relations' => $cgRelations
+                    ],
+                    'roles_mapping' => [
+                        'is_enabled' => $rolesMapping['is_enabled'] ? true : false,
+                        'apply_only_first_role' => $rolesMapping['apply_only_first_role'] ? true : false,
+                        'relations' => $aclRelations
+                    ],
+                    'introspection_token_endpoint' => ($customConfiguration['introspection_token_endpoint']
+                        ? true : false),
+                    'userinfo_endpoint' => ($customConfiguration['userinfo_endpoint'] ? true : false),
+                    'endsession_endpoint' => ($customConfiguration['endsession_endpoint'] ? true : false),
+                    'connection_scopes' => isset($customConfiguration['connection_scopes'])
+                        ? count($customConfiguration['connection_scopes']) : 0,
+                    'authentication_type' => $customConfiguration['authentication_type'],
+                    'verify_peer' => ($customConfiguration['verify_peer'] ? true : false),
+                    'auto_import' => ($customConfiguration['auto_import'] ? true : false)
+                ];
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * get configured authentications options
+     *
+     * @return array
+     */
+    public function getAuthenticationOptions()
+    {
+        $data = array_merge(
+            ['LDAP' => $this->getLDAPAuthenticationOptions()],
+            [$this->getNewAuthenticationOptions()][0]
+        );
```

Comment:
What do you want to do ? (a) adding a LDAP key if not set, or (b) real technical merging ?

If (a), I suggest one of these two syntaxes : 

1) express the logic via a "if"
```
$data = [$this->getNewAuthenticationOptions()][0];
if (!isset($data['LDAP'])) {
    $data['LDAP'] = $this->getLDAPAuthenticationOptions();
}
// FYI, php 8.1 above "if" syntax sugar equivalent we will be able to use one day
// $data['LDAP'] ??= $this->getLDAPAuthenticationOptions();
```

2) use the php `+` for arrays
```
$data = [$this->getNewAuthenticationOptions()][0] 
    + ['LDAP' => $this->getLDAPAuthenticationOptions()];
```

Reply:
Note: my proposals implicitly means that calling `$this->getLDAPAuthenticationOptions()` is not mandatory for this function.

If not, this is probably a very bad code pattern and need comments for that.